src/tool/xmeta/cpp/models/namespace_model.cpp

```diff
@@ -0,0 +1,120 @@
+#if defined(_WIN32)
+#include <string.h>
+#endif
+
+#include <string_view>
+
+#include "namespace_model.h"
+
+#include "using_directive_models.h"
+
+#include "class_model.h"
+#include "delegate_model.h"
+#include "enum_model.h"
+#include "interface_model.h"
+#include "property_model.h"
+#include "struct_model.h"
+
+namespace xlang::xmeta
+{
+    void namespace_body_model::add_class(std::shared_ptr<class_model> const& cm)
+    {
+        m_classes.insert(std::pair(std::string_view{ cm->get_id() }, cm));
```

Comment:
Seems like we should check the return value here so we can throw an error if a collision occurred.